Model/ProductProtection.php

```diff
@@ -358,6 +358,20 @@ private function upsert(
                 return;
             }
 
+            foreach ($quote->getItems() as $quoteItem) {
+                if (
+                    $quoteItem->getSku(Extend::WARRANTY_PRODUCT_SKU) &&
+                    $quoteItem->getOptionByCode('plan_id') &&
+                    $quoteItem->getOptionByCode('plan_id')->getValue() == $planId &&
+                    $quoteItem->getOptionByCode('associated_product_sku') &&
+                    $quoteItem->getOptionByCode('associated_product_sku')->getValue() == $productId
+                ) {
+                    $quoteItem->setQty($quoteItem->getQty() + $quantity);
```

Comment:
later on in this function, isn't
```
if (isset($quantity)) {
      $item->setQty($quantity);
  }
```
going to overwrite the incremented (original + new) quantity set here with the `$quantity` argument? if that's the case, would it make sense to just `save($quote);` and also `save($quote-collectTotals())` here, then `return` instead of `break` in this loop? i could be way off, let me know what you think.